File: apps/mobile/android/wear/src/main/kotlin/com/upcoach/wear/data/HabitModels.kt
package com.upcoach.wear.data

import com.google.gson.annotations.SerializedName
import java.util.Date

/**
 * Habit summary model synced from phone
 */
data class HabitSummary(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("emoji") val emoji: String?,
    @SerializedName("isCompletedToday") val isCompletedToday: Boolean,
    @SerializedName("currentStreak") val currentStreak: Int,
    @SerializedName("targetFrequency") val targetFrequency: Int,
    @SerializedName("completedThisWeek") val completedThisWeek: Int,
    @SerializedName("category") val category: String?,
    @SerializedName("color") val color: String?
) {
    val displayEmoji: String
        get() = emoji ?: "✓"
}

/**
 * Companion data synced from phone app
 */
data class CompanionData(
    @SerializedName("habits") val habits: List<HabitSummary> = emptyList(),
    @SerializedName("currentStreak") val currentStreak: Int = 0,
    @SerializedName("bestStreak") val bestStreak: Int = 0,
    @SerializedName("streakLastActivity") val streakLastActivity: Date? = null,
    @SerializedName("todayCompletedHabits") val todayCompletedHabits: Int = 0,
    @SerializedName("todayTotalHabits") val todayTotalHabits: Int = 0,
    @SerializedName("todayProgress") val todayProgress: Double = 0.0,
    @SerializedName("pendingHabitIds") val pendingHabitIds: List<String> = emptyList(),
    @SerializedName("weeklyProgress") val weeklyProgress: Double = 0.0,
    @SerializedName("totalPoints") val totalPoints: Int = 0,
    @SerializedName("currentLevel") val currentLevel: Int = 1,
    @SerializedName("lastUpdated") val lastUpdated: Date = Date(),
    @SerializedName("syncVersion") val syncVersion: Int = 1
) {
    val pendingHabits: List<HabitSummary>
        get() = habits.filter { !it.isCompletedToday }

    val completedHabits: List<HabitSummary>
        get() = habits.filter { it.isCompletedToday }

    val todayProgressPercent: Int
        get() = (todayProgress * 100).toInt()
}

/**
 * Habit completion request to send to phone
 */
data class HabitCompletionRequest(
    @SerializedName("habitId") val habitId: String,
    @SerializedName("completedAt") val completedAt: Date = Date(),
    @SerializedName("source") val source: String = "wearOS"
)

/**
 * Watch message types
 */
enum class WatchMessageType {
    @SerializedName("sync_request") SYNC_REQUEST,
    @SerializedName("habit_completed") HABIT_COMPLETED,
    @SerializedName("habit_uncompleted") HABIT_UNCOMPLETED,
    @SerializedName("refresh_data") REFRESH_DATA
}

/**
 * Watch message wrapper
 */
data class WatchMessage(
    @SerializedName("type") val type: WatchMessageType,
    @SerializedName("payload") val payload: String? = null,
    @SerializedName("timestamp") val timestamp: Date = Date()
)

File: apps/mobile/ios/UpCoachWatch/Models/HabitModels.swift
import Foundation

/// Habit summary model for watch display
struct HabitSummary: Codable, Identifiable {
    let id: String
    let name: String
    let emoji: String?
    let isCompletedToday: Bool
    let currentStreak: Int
    let targetFrequency: Int
    let completedThisWeek: Int
    let category: String?
    let color: String?

    /// Category icon based on habit type
    var categoryIcon: String {
        switch category?.lowercased() {
        case "health": return "heart.fill"
        case "fitness": return "figure.run"
        case "mindfulness": return "brain.head.profile"
        case "productivity": return "checkmark.circle.fill"
        case "learning": return "book.fill"
        case "social": return "person.2.fill"
        default: return "star.fill"
        }
    }

    /// Display emoji or default icon
    var displayEmoji: String {
        emoji ?? "✓"
    }
}

/// Companion data synced from phone
struct CompanionData: Codable {
    let habits: [HabitSummary]
    let currentStreak: Int
    let bestStreak: Int
    let streakLastActivity: Date?
    let todayCompletedHabits: Int
    let todayTotalHabits: Int
    let todayProgress: Double
    let pendingHabitIds: [String]
    let weeklyProgress: Double
    let totalPoints: Int
    let currentLevel: Int
    let lastUpdated: Date
    let syncVersion: Int

    /// Today's completion percentage as string
    var todayProgressText: String {
        return "\(Int(todayProgress * 100))%"
    }

    /// Number of pending habits
    var pendingCount: Int {
        return pendingHabitIds.count
    }

    /// Level display text
    var levelText: String {
        return "Level \(currentLevel)"
    }
}

/// Completion request to send back to phone
struct HabitCompletionRequest: Codable {
    let habitId: String
    let completedAt: Date
    let source: String

    init(habitId: String) {
        self.habitId = habitId
        self.completedAt = Date()
        self.source = "appleWatch"
    }
}

/// Message types for watch-phone communication
enum WatchMessageType: String, Codable {
    case syncRequest = "sync_request"
    case habitCompleted = "habit_completed"
    case habitUncompleted = "habit_uncompleted"
    case refreshData = "refresh_data"
}

/// Watch message wrapper
struct WatchMessage: Codable {
    let type: WatchMessageType
    let payload: String?
    let timestamp: Date

    init(type: WatchMessageType, payload: String? = nil) {
        self.type = type
        self.payload = payload
        self.timestamp = Date()
    }
}
